packages/p2p-v2/src/pages/my-ads/components/AdTypeSection/AdTypeSection.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { MouseEventHandler, useEffect } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { useHistory } from 'react-router-dom';
+import { RadioGroup } from '@/components';
+import { BUY_SELL, MY_ADS_URL, RATE_TYPE } from '@/constants';
+import { AdFormController } from '../AdFormController';
+import { AdFormInput } from '../AdFormInput';
+import { AdFormTextArea } from '../AdFormTextArea';
+import './AdTypeSection.scss';
+
+type TAdTypeSectionProps = {
+    currency: string;
+    getCurrentStep: () => number;
+    getTotalSteps: () => number;
+    goToNextStep: MouseEventHandler<HTMLButtonElement>;
+    goToPreviousStep: MouseEventHandler<HTMLButtonElement>;
+    localCurrency?: string;
+    rateType: string;
+};
+
+const AdTypeSection = ({ currency, localCurrency, rateType, ...props }: TAdTypeSectionProps) => {
+    const {
+        control,
+        formState: { isDirty, isValid },
+        getValues,
+        setValue,
+        trigger,
+        watch,
+    } = useFormContext();
+
+    const history = useHistory();
+    const isSell = watch('ad-type') === BUY_SELL.SELL;
+
+    // Trigger validation for all fields when any field changes
+    useEffect(() => {
+        trigger();
+    }, [getValues()]); // Trigger whenever getValues() changes
```

Comment:
Why not just use it in a variable?
```suggestion
  const values = getValues()
   useEffect(() => {
      ...
    }, [values]); // Trigger whenever getValues() changes
```